node/src/BaseClient.ts

```diff
@@ -456,15 +499,19 @@ export class BaseClient {
             const errorType = getRequestErrorClass(message.requestError.type);
             reject(new errorType(message.requestError.message ?? undefined));
         } else if (message.respPointer != null) {
-            const pointer = message.respPointer;
+            let pointer;
 
-            if (typeof pointer === "number") {
-                // TODO: change according to https://github.com/valkey-io/valkey-glide/pull/2052
-                resolve(valueFromSplitPointer(0, pointer, true));
+            if (typeof message.respPointer === "number") {
+                pointer = new PointerResponse(message.respPointer);
             } else {
-                // TODO: change according to https://github.com/valkey-io/valkey-glide/pull/2052
-                resolve(valueFromSplitPointer(pointer.high, pointer.low, true));
+                pointer = new PointerResponse(
```

Comment:
@barshaul We really need docs here, it's a hard part of the code to understand, we add docs to much simpler stuff.
This part will be hard to maintain if the ones who know what's going on would not be here.
@adarovadya I know It's not exactly your logic here, and you just built on the existing logic, but if you can, with the guidance of Bar to add some docs here it will help a lot.